From 395fb7877a140602e4247690ad02975259bcc6ab Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Mon, 22 Nov 2021 12:30:18 +0000
Subject: [PATCH] dtoverlays: Convert ov5647 to use the regulator
 framework

Fixing up shutdown GPIOs via overrides is ugly, and doesn't work
on eg CM4 where both cameras share the same shutdown GPIO.

The driver is now updated to use the regulator framework, so switch
to using that instead.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/ov5647-overlay.dts | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

--- a/arch/arm/boot/dts/overlays/ov5647-overlay.dts
+++ b/arch/arm/boot/dts/overlays/ov5647-overlay.dts
@@ -18,9 +18,12 @@
 				reg = <0x36>;
 				status = "okay";
 
-				pwdn-gpios = <&gpio 41 1>, <&gpio 32 1>;
 				clocks = <&cam1_clk>;
 
+				avdd-supply = <&cam1_reg>;
+				dovdd-supply = <&cam_dummy_reg>;
+				dvdd-supply = <&cam_dummy_reg>;
+
 				rotation = <0>;
 				orientation = <2>;
 
@@ -67,12 +70,9 @@
 	};
 
 	fragment@4 {
-		target-path="/__overrides__";
+		target = <&cam1_reg>;
 		__overlay__ {
-			cam0-pwdn-ctrl = <&ov5647>,"pwdn-gpios:0";
-			cam0-pwdn      = <&ov5647>,"pwdn-gpios:4";
-			cam0-led-ctrl  = <&ov5647>,"pwdn-gpios:12";
-			cam0-led       = <&ov5647>,"pwdn-gpios:16";
+			startup-delay-us = <20000>;
 		};
 	};
 
